### PR TITLE
Make ``Module`` call the ``__init__`` of ``NodeNG``

### LINKED ISSUE #1273: ``nodes.Module`` don't have a ``end_lineno`` and ``end_col_offset``
### Steps to reproduce

```python
import astroid

code = """
    print("a module")
    """

module = astroid.parse(code)
print(module.end_lineno)
print(module.end_col_offset)
```

### Current behavior

`AttributeError` on both of the last lines.

### Expected behavior

@cdce8p Let me know if I misunderstood you, but I thought we wanted these to be accessible on all nodes, just initialised as `None`.
If that was not the case, I would make the case to do so as it allows you to do `node.end_lineno` without running in to `AttributeError`'s.

### Version

Latest `main`.

### PR DESCRIPTION
**Pull request by DanielNoord**
## Steps

- [x] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description

Found this while working on the `end_col_offset` issues.

@cdce8p Do you know why we don't do this? Without this `col_offset` isn't defined on `nodes.Module`. We could also define it in the `__init__` of `nodes.Module`, but this seem elegant as well?

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Related Issue

Closes #1273

### CODE REVIEW

**Comment by DanielNoord** [comment]
I guess I know why know: it will set `node.lineno` to `None` and we make many unguarded calls to `node.lineno` in `pylint`. We tend to do `int() + node.lineno` which shouldn't be allowed if we know that `node.lineno` can be `None`.

Problem for the future! Sorry for pinging!

File: astroid/nodes/scoped_nodes.py [hunk 1]
```diff
@@ -524,6 +523,8 @@ def __init__(
         """
         self.future_imports = set()
 
+        super().__init__(lineno=None, col_offset=None, parent=parent)
```
Comment (cdce8p) [inline astroid/nodes/scoped_nodes.py:526]:
```suggestion
        super().__init__(parent=parent)
```
That is enough

**Review by cdce8p — COMMENTED** [review]
> Do you know why we don't do this?

Unfortunately not.

> Without this `col_offset` isn't defined on `nodes.Module`.

Haven't though about this tbh, but you're right. It should be defined even if `None`. That's what all other nodes without `lineno` and `col_offset` do too: https://github.com/PyCQA/astroid/commit/40d556070fd566b7bb43cb3ed0e498045b88f2d0

File: astroid/nodes/scoped_nodes.py [hunk 2]
```diff
@@ -524,6 +523,8 @@ def __init__(
         """
         self.future_imports = set()
 
+        super().__init__(parent=parent)
```
Comment (cdce8p) [inline astroid/nodes/scoped_nodes.py:526]:
```suggestion
        super().__init__(lineno=1, parent=parent)
```
Mentioned it here: https://github.com/PyCQA/astroid/issues/1273#issuecomment-975510511. Let's try if that works.

File: astroid/nodes/scoped_nodes.py [hunk 3]
```diff
@@ -524,6 +523,8 @@ def __init__(
         """
         self.future_imports = set()
 
+        super().__init__(lineno=1, parent=parent)
```
Comment (cdce8p) [inline astroid/nodes/scoped_nodes.py:526]:
```suggestion
        super().__init__(lineno=0, parent=parent)
```
Seems like the `lineno` needs to be `0`.

File: tests/unittest_nodes_lineno.py [hunk 4]
```diff
@@ -1221,3 +1222,14 @@ class X(Parent, var=42):
         assert (c1.body[0].lineno, c1.body[0].col_offset) == (4, 4)
         assert (c1.body[0].end_lineno, c1.body[0].end_col_offset) == (4, 8)
         # fmt: on
+
+    @staticmethod
+    def test_end_lineno_module() -> None:
+        """Test end_lineno and end_col_offset for Module"""
```
Comment (DanielNoord) [inline tests/unittest_nodes_lineno.py:1228]:
```suggestion
        """Tests for Module"""
```

File: tests/unittest_nodes_lineno.py [hunk 5]
```diff
@@ -1221,3 +1222,14 @@ class X(Parent, var=42):
...
         assert (c1.body[0].end_lineno, c1.body[0].end_col_offset) == (4, 8)
         # fmt: on
+
+    @staticmethod
+    def test_end_lineno_module() -> None:
+        """Tests for Module"""
+        code = """print()"""
+        module = astroid.parse(code)
+        assert isinstance(module, nodes.Module)
+        assert module.lineno == 1
```
Comment (DanielNoord) [inline tests/unittest_nodes_lineno.py:1232]:
```suggestion
        assert module.lineno == 0
```

**Review by Pierre-Sassoulas — APPROVED** [review]
LGTM, do you want another review @DanielNoord ?

**Comment by DanielNoord** [comment]
Let's wait for @cdce8p. He reported the initial issue that prompted this PR.

File: astroid/nodes/scoped_nodes.py [hunk 6]
```diff
@@ -389,10 +389,8 @@ class Module(LocalsDictNodeNG):
 
     :type: int or None
     """
-    lineno = 0
+    lineno: Literal["0"] = 0
```
Comment (cdce8p) [inline astroid/nodes/scoped_nodes.py:392]:
```suggestion
    lineno: Literal[0] = 0
```

**Review by cdce8p — CHANGES_REQUESTED** [review]
Just some small comments, looks good otherwise.

**Review by cdce8p — APPROVED** [review]

**Pull request merged**